programs/clearing_house/src/controller/position.rs

```diff
@@ -65,116 +61,143 @@ pub fn get_position_index(
 }
 
 pub struct PositionDelta {
-    pub quote_asset_amount: u128,
+    pub quote_asset_amount: i128,
     pub base_asset_amount: i128,
 }
 
+enum PositionUpdateType {
+    Open,
+    Increase,
+    Reduce,
+    Close,
+    Flip,
+}
+
+fn get_position_update_type(
+    position: &MarketPosition,
+    delta: &PositionDelta,
+) -> PositionUpdateType {
+    if position.base_asset_amount == 0 {
+        PositionUpdateType::Open
+    } else if position.base_asset_amount.signum() == delta.base_asset_amount.signum() {
+        PositionUpdateType::Increase
+    } else if position.base_asset_amount.abs() > delta.base_asset_amount.abs() {
+        PositionUpdateType::Reduce
+    } else if position.base_asset_amount.abs() == delta.base_asset_amount.abs() {
+        PositionUpdateType::Close
+    } else {
+        PositionUpdateType::Flip
```

Comment:
when delta.baa == 0 and position.baa != 0 it goes for a flip ? idk if that would ever be hit tho - might be good to have an explicit case or throw an error?

Reply:
yeah we should check that the delta is non-zero, good idea